central/debug/service/service.go

```diff
@@ -494,6 +495,11 @@ type debugDumpOptions struct {
 	withCentral       bool
 	clusters          []string
 	since             time.Time
+	// timeout value is sent by client. We operate under assumption that after timeout,
+	// client will drop connection, potentially resulting in incorrect diagnostic bundle.
+	// To avoid that, long-running operations(getting k8s diangostics, pulling sensor metrics) can be dropped
+	// if time is running out.
+	timeout time.Duration
```

Comment:
I don't necessarily like setting the timeout now explicitly within a query parameter.

Initially, I would expect the timeout to come from the HTTP request settings, e.g. re-using the `http.Request.Context()` cancellation and not specifically handle timeouts within our server side logic but rather offload it to the client (like we do within roxctl by re-using `request.WithContext(context.WithTimeout(timeout))`.

Meaning, we continue to try to assemble the diagnostic bundle without _actually_ setting a specific, overarching timeout, but rather verifying whether the request's context has been cancelled or not.

You could change the condition of invoking `writeTimeoutWarning` to `ctx.Err() != nil` (however, writing _anything_ to the response writer / zip writer won't be sent to the client, so we'd change to logging the instance in this case and returning the function).

For this to work and us not ending up having blocking reads on channels when receiving e.g. messages from sensor, we would have to create some sort of internal timeout based on the longest amount of time we would be waiting for a sensor message (this would be something we'd have to find a median).

This would make the API feel more native in my opinion and we could decide to print the warning message you have already created within e.g. `roxctl` when the API call returns a `context.DeadlineExceeded` error (which is what you will get), and decide to have no explicit timeout being set within the UI (but relying on the internal timeouts to not end up in e.g. a "infinite waiting on messages from a channel" situation).

(just for posterity, the following test code highlights what happens when a HTTP request's timeout is hit on both client and server side):
```go
	server := httptest.NewServer(http.HandlerFunc(func(writer http.ResponseWriter, request *http.Request) {
		time.Sleep(2 * time.Second)
		assert.Error(t, request.Context().Err())
		assert.Error(t, request.Context().Err(), context.Canceled)
		return
	}))
	defer server.Close()

	ctx, _ := context.WithTimeout(context.Background(), 1*time.Second)
	req, err := http.NewRequestWithContext(ctx, http.MethodGet, server.URL, nil)
	assert.NoError(t, err)

	client := http.Client{}
	resp, err := client.Do(req)
	assert.Error(t, err)
	assert.ErrorIs(t, err, context.DeadlineExceeded)
	assert.Nil(t, resp)
        time.Sleep(2 * time.Second)
```